src/plugins/share/server/routes/goto.ts

```diff
@@ -65,7 +67,15 @@ export const createGotoRoute = ({
 
         const prependedUrl = modifyUrl(url, (parts) => {
           if (!parts.hostname && parts.pathname && parts.pathname.startsWith('/')) {
-            parts.pathname = `${basePath}${parts.pathname}`;
+            if (getWorkspaceState(request).requestWorkspaceId) {
+              parts.pathname = getRedirectUrl({
+                request,
+                nextUrl: parts.pathname,
+                basePath,
+              });
+            } else {
+              parts.pathname = `${basePath}${parts.pathname}`;
+            }
           }
```

Comment:
nit: could remove the if condition as `getRedirectUrl` have same logic inside.

Reply:
Was trying to introduce less regression risk.